### PR TITLE
Hamcrest and Junit references added

### PR DESCRIPTION
**Pull request by lukepadlp**
Added Hamcrest (all, 1.3) and JUnit (4.12) references into the .iml
files of all the apps (along with the installer and core) so the
matchers and test code (especially the stuff in core/java/test/junit)
could run.

Edit: Previously, the .iml files had no references to hamcrest, and in some cases JUnit too, which would pose difficulties/errors for testing and running i2p (depending on the part of i2p being executed)

### CODE REVIEW

File: core/core.iml [hunk 1]
```diff
@@ -10,6 +10,20 @@
...
-    <orderEntry type="library" scope="TEST" name="org.hamcrest:hamcrest-library:1.3" level="application" />
+    <orderEntry type="library" scope="TEST" name="org.hamcrest:hamcrest-all:1.3" level="application" />
+    <orderEntry type="library" name="lib" level="project" />
+    <orderEntry type="library" name="tomcat-coyote-util" level="project" />
+    <orderEntry type="library" name="start" level="project" />
+    <orderEntry type="library" name="javax.servlet.jsp-2.2.0.v201112011158" level="project" />
+    <orderEntry type="module-library" scope="TEST">
+      <library>
+        <CLASSES>
+          <root url="jar://$MODULE_DIR$/java/test/junit/net/i2p/junit-4.12.jar!/" />
```
Comment (lukepadlp) [inline core/core.iml]: This line (21), and the one below it (22), are examples of where a reference to JUnit and hamcrest were left out (no .jar files or supporting libraries of any form were linked, so matchers were not working and tests wouldn't run).

**Comment by nextl00p** [comment]
This repository is updated only by an hourly cronjob which exports our monotone repository. We will have to apply this manually using diff probably. See https://geti2p.net/en/get-involved/guides/new-developers#git

**Comment by majestrate** [comment]
java i2p uses monotone inside of the i2p network for their main development code

github is a mirror of that.

**Comment by nextl00p** [comment]
I am going to squash your two commits and prepare a patch and submit it into our bugtracker.

**Comment by nextl00p** [comment]
https://trac.i2p2.de/ticket/1891

**Comment by lukepadlp** [comment]
Understood, majestrate. I will be using monotone for i2p contributions from now on.

Thanks, nextl00p!

**Comment by str4d** [comment]
Sorry for not addressing this sooner! I have actually just removed the .iml files, because we now have a Gradle build script for generating them (`./gradlew idea`). Please test that, and open a new issue if you still have problems 🙂

**Pull request closed without merging**